src/connection/activate.ts

```diff
@@ -0,0 +1,88 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceEventName, WalletConnectionResult } from '@uniswap/analytics-events'
+import { Connection } from 'connection'
+import { atom } from 'jotai'
+import { useAtomValue, useUpdateAtom } from 'jotai/utils'
+import { useCallback } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import { updateSelectedWallet } from 'state/user/reducer'
+
+import { didUserReject } from './utils'
+
+export enum ActivationStatus {
+  PENDING,
+  ERROR,
+  IDLE,
+}
+
+type ActivationPendingState = { status: ActivationStatus.PENDING; connection: Connection }
+type ActivationErrorState = { status: ActivationStatus.ERROR; connection: Connection; error: any }
+const IDLE_ACTIVATION_STATE = { status: ActivationStatus.IDLE } as const
+
+type ActivationState = ActivationPendingState | ActivationErrorState | typeof IDLE_ACTIVATION_STATE
+
+const pendingConnectionStateAtom = atom<ActivationState>(IDLE_ACTIVATION_STATE)
```

Comment:
Why is this _pending_, and also why does the name mismatch the type?

Why not `activationStateAtom`?

Reply:
neglected to update after renaming the type, ty for catching